artifactory/commands/transferfiles/state/statemanager.go

```diff
@@ -0,0 +1,250 @@
+		repo, err := state.getRepository(repoKey, false)
+		if err != nil {
+			return err
+		}
+		repo.TransferredSizeBytes += totalSizeInBytes
+		repo.TransferredUnits += totalFiles
+		return nil
+	})
+	if err != nil {
+		return err
+	}
+	return ts.TransferRunStatus.action(func(transferRunStatus *TransferRunStatus) error {
+		transferRunStatus.TransferredSizeBytes += totalSizeInBytes
+		return nil
+	})
+}
+
+// Adds new diff details to the repo's diff array in state.
+// Marks files handling as started, and sets the handling range.
+func (ts *TransferStateManager) AddNewDiffToState(repoKey string, startTime time.Time) error {
+	return ts.TransferState.action(func(state *TransferState) error {
+		repo, err := state.getRepository(repoKey, false)
+		if err != nil {
+			return err
+		}
+		newDiff := DiffDetails{}
+
+		// Set Files Diff Handling started.
+		newDiff.FilesDiffRunTime.Started = ConvertTimeToRFC3339(startTime)
+
+		// Determines the range on which files diffs should be handled.
+		// If the repo previously completed files diff phases, we will continue handling diffs from where the last phase finished handling.
+		// Otherwise, we will start handling diffs from the start time of the full transfer.
+		for i := len(repo.Diffs) - 1; i >= 0; i-- {
+			if repo.Diffs[i].Completed {
+				newDiff.HandledRange.Started = repo.Diffs[i].HandledRange.Ended
+				break
+			}
+		}
+		if newDiff.HandledRange.Started == "" {
+			newDiff.HandledRange.Started = repo.FullTransfer.Started
+		}
+		newDiff.HandledRange.Ended = ConvertTimeToRFC3339(startTime)
+		repo.Diffs = append(repo.Diffs, newDiff)
+		return nil
+	})
+}
+
+func (ts *TransferStateManager) SetFilesDiffHandlingCompleted(repoKey string) error {
+	return ts.TransferState.action(func(state *TransferState) error {
+		repo, err := state.getRepository(repoKey, false)
+		if err != nil {
+			return err
+		}
+		repo.Diffs[len(repo.Diffs)-1].FilesDiffRunTime.Ended = ConvertTimeToRFC3339(time.Now())
+		repo.Diffs[len(repo.Diffs)-1].Completed = true
+		return nil
+	})
+}
+
+func (ts *TransferStateManager) GetReposTransferredSizeBytes(transferredSizeBytes *int64, repoKeys ...string) error {
+	return ts.TransferState.action(func(state *TransferState) error {
+		reposSet := datastructures.MakeSet[string]()
+		for _, repoKey := range repoKeys {
+			reposSet.Add(repoKey)
+		}
+		*transferredSizeBytes = 0
+		for i := range state.Repositories {
+			if reposSet.Exists(state.Repositories[i].Name) {
+				(*transferredSizeBytes) += state.Repositories[i].TransferredSizeBytes
+			}
+		}
+		return nil
+	})
+}
+
+func (ts *TransferStateManager) GetTransferredSizeBytes(transferredSizeBytes *int64) error {
+	return ts.TransferRunStatus.action(func(transferRunStatus *TransferRunStatus) error {
+		*transferredSizeBytes = transferRunStatus.TransferredSizeBytes
+		return nil
+	})
+}
+
+func (ts *TransferStateManager) GetDiffHandlingRange(repoKey string, start, end *time.Time) error {
```

Comment:
This comment applies to all getters in this struct, and not only this one.
I suggest we add the value this function retrieves as a returned value of the function. Inn other words, let's change the signature of this function to be - 
```
func (ts *TransferStateManager) GetDiffHandlingRange(repoKey string, start) (end time.Time, err error) {
```